icu4c/source/i18n/units_complexconverter.cpp

```diff
@@ -105,11 +107,21 @@ UBool ComplexUnitsConverter::greaterThanOrEqual(double quantity, double limit) c
     return newQuantity >= limit;
 }
 
-MaybeStackVector<Measure> ComplexUnitsConverter::convert(double quantity, UErrorCode &status) const {
+MaybeStackVector<Measure> ComplexUnitsConverter::convert(double quantity,
+                                                         icu::number::impl::RoundingImpl *rounder,
+                                                         UErrorCode &status) const {
     // TODO(icu-units#63): test negative numbers!
     // TODO(hugovdm): return an error for "foot-and-foot"?
     MaybeStackVector<Measure> result;
 
+    // For N converters:
+    // - the first converter converts from the input unit to the largest unit,
+    // - N-1 converters convert to bigger units for which we want integers,
+    // - the Nth converter (index N-1) converts to the smallest unit, for which
+    //   we keep a double.
+    MaybeStackArray<int64_t, 3> intValues(unitConverters_.length() - 1, status);
```

Comment:
I'm making it 5 now, which is sufficient for a mixed unit of length 6. (int64 * 5 = 40 bytes.)

On the one hand, "might as well make it 8", because 64 bytes isn't a lot in this day and age. On the other, a mixed unit with length 6 is already feeling "pathological" relative to their intended purpose? :-)